docs/tutorials/3_The_Part_In_Between.rst

```diff
@@ -0,0 +1,196 @@
+see people aging as time passes.
+
+Using (Fake) Real Data
+----------------------
+
+Alright, now people are getting older but that doesn't really mean
+much because our mortality rate is static. What we want is for the
+mortality rate to depend on properties of the simulants. For now our
+only properties are age and sex so we'll start with those. If you
+happen to have a csv file of realistic mortality rates by age and sex
+lying around (and if you're the kind of person who's interested in
+Vivarium, you probably do) then use that. If not, you can make one
+up. It's easy:
+
+.. code-block:: python
+
+    import pandas as pd
+    import numpy as np
+
+    rows = []
+    for year in range(1990, 2015):
+            for age in range(150):
+                for sex in ['Female', 'Male']:
+                    rows.append([year, age, sex, np.random.random()])
+    df = pd.DataFrame(rows, columns=['year', 'age', 'sex', 'rate'])
+    df.to_csv('mortality_rate.csv', index=False)
+
+That just loops through all the permutations of age and sex that we
+might need and assigns a random number to each dumps it to disk where
+we can read it later. Obviously, in a real model you'd want to get the
+data from a more reputable source than a random number generator.
+
+With this reference data we can go back to our mortality code and get
+rid of that static rate. The first step is to register the reference
+data with the simulation. This isn't strictly necessary but having the
+simulation manage the data for us lets it do some optimizations that
+make access faster than it would be if we just loaded a pandas
+dataframe and used it directly. First we go to the ``Mortality``
+class's ``setup`` method and tweak it to load in the mortality rate
+table:
+
+.. code-block:: python
+
+    def setup(self, builder):
+        self.mortality_rate = builder.rate('mortality_rate')
+        self.rate_table = builder.lookup(pd.read_csv('mortality_rate.csv'))
+
+The ``builder.lookup`` method takes a ``DataFrame`` and loads it into
+the internal reference data system returning a function that can be
+used to access it by population index later (the same indexes we've
+been using to access and update the population table). The
+``key_columns`` parameter specifies which simulant attributes should
+be used to look up the data, in this case our data is indexed by age
+and sex so we use those [#]_. Using it later is as simple as calling
+``self.rate_table``:
+
+.. code-block:: python
+
+    self.rate_table(index)
+
+That will return the rate corresponding to each simulant in the
+index. We can use that in our model by putting it in place of the
+static rate we had before:
+
+.. code-block:: python
+
+    @produces_value('mortality_rate')
+    def base_mortality_rate(self, index):
+        return self.rate_table(index)
+
+And that's it. If you run the simulation people will die according to
+the new dynamic mortality rate.
```

Comment:
`dynamic` is an ineffective word for describing what we're actually doing.  The mortality rate is parameterized by demography, if it's built that way.